vm/src/cformat.rs

```diff
@@ -412,6 +561,159 @@ impl CFormatPart {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub(crate) struct CFormatBytes {
+    parts: Vec<(usize, CFormatPart)>,
```

Comment:
CFormatPart should probably be generic to allow `String` or `Vec<u8>`

Reply:
I'm not sure that is necessary, I am using String as Vec<u8> here for bytes. As String stores char as utf-8 so I don't think any cost here.

Reply:
@coolreader18 do you have an example which distinguish bytes and str for the case?

Reply:
Well bytes can have non-ascii and still be formatted, while a String wouldn't conserve that:

```py
>>> b'\xaa %b' % b'123'
b'\xaa 123'
```

Specifically, the `u8 as char` casting in this PR would map the non-ascii byte `\xaa` to the utf8 character sequence `'ª' == b'\xc2\xaa'`